lib/charms/alertmanager_k8s/v0/alertmanager_remote_configuration.py

```diff
@@ -0,0 +1,1890 @@
+
+        Uses JSON Schema validator to check whether the Alertmanager configuration provided
+        by the `alertmanager-remote-configuration` consumer is valid.
+
+        Args:
+            config: Alertmanager configuration JSON
+
+        Returns:
+            bool: True/False depending on the configuration validity
+        """
+        try:
+            validate(instance=config, schema=ALERTMANAGER_CONFIG_JSON_SCHEMA)
+            return True
+        except exceptions.ValidationError:
+            return False
+
+    def _get_current_config(self) -> str:
+        """Gets current configuration of the Alertmanager.
+
+        Uses Alertmanager's API to get the current configuration.
+
+        Returns:
+            str: Alertmanager configuration
+        """
+        config_endpoint = "api/v2/status"
+        url = "{}/{}".format(self.api_address, config_endpoint)
+        response = requests.get(url)
+        return response.json()["config"]["original"] if response.status_code == 200 else ""
+
+
+class AlertmanagerRemoteConfigurationConsumer(Object):
+    """API that manages a required `alertmanager_remote_configuration` relation.
+
+    The `AlertmanagerRemoteConfigurationConsumer` is intended to be used by charms that need
+    to push data to other charms over the `alertmanager_remote_configuration` interface.
+
+    The `AlertmanagerRemoteConfigurationConsumer` object can be instantiated as follows
+    in your charm:
+
+    ```
+    from charms.alertmanager_k8s.v0.alertmanager_remote_configuration import
+        AlertmanagerRemoteConfigurationConsumer,
+    )
+
+    def __init__(self, *args):
+        ...
+        self.remote_configuration_consumer = AlertmanagerRemoteConfigurationConsumer(self)
+        ...
+    ```
+
+    The `AlertmanagerRemoteConfigurationConsumer` assumes that, in the `metadata.yaml`
+    of your charm, you declare a required relation as follows:
+
+    ```
+    requires:
+        remote-configuration:  # Relation name
+            interface: alertmanager_remote_configuration  # Relation interface
+    ```
+
+    The `AlertmanagerRemoteConfigurationConsumer` provides handling of the most relevant charm
+    lifecycle events. On each of the defined Juju events, Alertmanager configuration and templates
+    from a specified file will be pushed to the relation data bag.
+    Inside the relation data bag, Alertmanager configuration will be stored under
+    `alertmanager_configuration` key, while the templates under the `alertmanager_templates` key.
+    Separation of the main configuration and the templates is dictated by the assumption that
+    the default provider of the `alertmanager_remote_configuration` relation will be
+    `alertmanager-k8s` charm, which requires such separation.
+    """
+
+    def __init__(
+        self,
+        charm: CharmBase,
+        relation_name: str = DEFAULT_RELATION_NAME,
+        config_file_path: str = DEFAULT_ALERTMANAGER_CONFIG_FILE_PATH,
```

Comment:
Not sure why the lib needs access to the config file.

Reply:
It doesn't anymore.
Removed.